### PR TITLE
feat: added user/me end point

### PR DESCRIPTION
**Pull request by duypv98**

### CODE REVIEW

File: middlewares/verifyUser.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+function verifyUser(req, res, next) {
+  if (!req.uid) {
+    throw new UnauthorizedError();
+  }
+  let reqUserId = req.params.userId;
+  if (reqUserId === 'me') {
+    reqUserId = req.uid;
+  } else if (reqUserId !== req.uid) {
+    throw new PermissionDeniedError();
+  }
```
Comment (namdaoduy) [inline middlewares/verifyUser.js]:
```
if (req.params.userId !== me && req.params.userId !== req.uid) {
  throw ...
}
```

File: controllers/users.controller.js [hunk 2]
```diff
@@ -2,7 +2,8 @@ import asyncDec from 'utils/asyncDecoration';
 import userCore from 'core/users.core';
 
 const getUserById = asyncDec(async (req, res) => {
-  const dataResponse = await userCore.findUserById(req.params.userId);
+  const userId = (req.params.userId === 'me') ? req.uid : req.params.userId;
+  const dataResponse = await userCore.findUserById(userId);
```
Comment (namdaoduy) [inline controllers/users.controller.js]: Remove and only use req.uid

**Review by namdaoduy — CHANGES_REQUESTED** [review]

File: common/error.js [hunk 3]
```diff
@@ -95,6 +95,16 @@ export class TokenExpiredError extends ServerAPIError {
...
+export class PermissionDeniedError extends ServerAPIError {
+  constructor({ ...payload }) {
+    super({
+      code: 40303,
+      message: 'Permission Denied',
+      ...payload,
+    });
+  }
+}
+
```
[the remark this answers is not in the document]
Comment (duypv98) [reply common/error.js]: https://github.com/EasyEventV2/ee2_api/pull/13/commits/665b358ab4ebc2b0f702f73adaeece6bad79de02

**Review by namdaoduy — APPROVED** [review]

File: controllers/users.controller.js (hunk 2 above)
Comment (duypv98) [reply controllers/users.controller.js]: https://github.com/EasyEventV2/ee2_api/pull/13/commits/7a32e97e8ab7d658de30835a436504293227f42b

File: middlewares/verifyUser.js (hunk 1 above)
Comment (duypv98) [reply middlewares/verifyUser.js]: https://github.com/EasyEventV2/ee2_api/pull/13/commits/1667941e518aee4016ce96f3bbeec0f6ef59ff06

**Pull request merged**